cpp/tensorrt_llm/kernels/unfusedAttentionKernels/unfusedAttentionKernels_2_template.h

```diff
@@ -800,6 +801,10 @@ __global__ void applyBiasRopeUpdateKVCacheV2(QKVPreprocessingParams<T, KVCacheBu
     // Make sure there are multiple of tokens_per_block otherwise syncthreads will lead to deadlocks.
     int const tokens_loop_end = int((params.token_num + TOKENS_PER_BLOCK - 1) / TOKENS_PER_BLOCK) * TOKENS_PER_BLOCK;
 
+#if (defined(__CUDA_ARCH__) && (__CUDA_ARCH__ >= 900))
```

Comment:
Which member of `params` is produced by the previous kernel? Looks like potentially `tokens_info`, `cu_seq_lens ` and  `token_num`? If so, this would affect correctness. Please put `griddepcontrol.wait` at line 782 then (after `int const head_idx = blockIdx.y;`).